service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredComponentTest.java

```diff
@@ -0,0 +1,136 @@
+        String storageDataWrapperString = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+
+        var storageDataWrapper = OBJECT_MAPPER.readValue(storageDataWrapperString, StorageDataWrapper.class);
+        assertThatObjectCreated(storageDataWrapper, ehrExtractUpdated, structuredTaskDefinition2);
+        assertThat(structuredTaskDefinition1.getTaskId()).isNotEqualTo(ehrExtractUpdated.getGpcAccessStructured().getTaskId());
+    }
+
+    @Test
+    public void When_StructuredTaskPatientNotFoundError_Expect_EhrStatusNotUpdatedAndNotSavedToStorage() {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatusRepository.save(ehrExtractStatus);
+
+        GetGpcStructuredTaskDefinition structuredTaskDefinition1 = buildInvalidNHSNumberStructuredTask(ehrExtractStatus);
+        Exception exception = assertThrows(RuntimeException.class, () -> {
+            getGpcStructuredTaskExecutor.execute(structuredTaskDefinition1);
+        });
+        assertThat(exception.getMessage()).isEqualTo(EXPECTED_ERROR_RESPONSE);
+    }
+
+    private GetGpcStructuredTaskDefinition buildValidStructuredTask(EhrExtractStatus ehrExtractStatus) {
+        return GetGpcStructuredTaskDefinition.builder()
+            .fromAsid(ehrExtractStatus.getEhrRequest().getFromAsid())
+            .toAsid(ehrExtractStatus.getEhrRequest().getToAsid())
+            .fromOdsCode(ehrExtractStatus.getEhrRequest().getFromOdsCode())
+            .conversationId(ehrExtractStatus.getConversationId())
+            .nhsNumber(ehrExtractStatus.getEhrRequest().getNhsNumber())
+            .requestId(ehrExtractStatus.getEhrRequest().getRequestId())
+            .taskId(UUID.randomUUID().toString())
+            .build();
+    }
+
+    private void assertThatInitialRecordWasUpdated(EhrExtractStatus ehrExtractStatusUpdated, EhrExtractStatus ehrExtractStatus) {
+        assertThat(ehrExtractStatusUpdated.getUpdatedAt()).isNotEqualTo(ehrExtractStatus.getUpdatedAt());
+        var gpcAccessStructured = ehrExtractStatusUpdated.getGpcAccessStructured();
+        assertThat(gpcAccessStructured.getObjectName()).isEqualTo(ehrExtractStatus.getConversationId() + GPC_STRUCTURED_FILE_EXTENSION);
+        assertThat(gpcAccessStructured.getAccessedAt()).isNotNull();
+        assertThat(gpcAccessStructured.getTaskId()).isNotNull();
+    }
+
+    private void assertThatObjectCreated(StorageDataWrapper storageDataWrapper, EhrExtractStatus ehrExtractStatus,
+        GetGpcStructuredTaskDefinition structuredTaskDefinition) {
+        assertThat(storageDataWrapper.getConversationId()).isEqualTo(ehrExtractStatus.getConversationId());
+        assertThat(storageDataWrapper.getTaskId()).isEqualTo(ehrExtractStatus.getGpcAccessStructured().getTaskId());
+        assertThat(storageDataWrapper.getType()).isEqualTo(structuredTaskDefinition.getTaskType().getTaskTypeHeaderValue());
+        assertThat(storageDataWrapper.getResponse()).isNotBlank();
```

Comment:
We need to find some way to verify that we're actually storing a structured record here. That is one of the reasons why we needed to write these tests. We were actually storing HTML instead.

I don't think we need to verify that it is the EXACT same response - especially since we depend on the public demo api to provide this. The data could change at any time.

It might be enough to check that it contains the FHIR profile: `https://fhir.nhs.uk/STU3/StructureDefinition/GPConnect-StructuredRecord-Bundle-1`

That is always included in the response

```
{
    "resourceType": "Bundle",
    "meta": {
        "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/GPConnect-StructuredRecord-Bundle-1"
        ]
    },
```

And only used for structured record responses.